libraries/hatch-server-user-management/src/UserManagementError.ts

```diff
@@ -0,0 +1,17 @@
+export enum UserManagementErrorCodes {
+  ACCOUNT_LOCKED = 'AccountLocked',
+  EXPIRED_TOKEN = 'ExpiredToken',
+  INVALID_CODE = 'InvalidCode',
+  INVALID_PASSWORD = 'InvalidPassword',
+  INTERNAL_ERROR = 'InternalError',
+  UNAUTHORIZED = 'Unauthorized',
+  USERNAME_EXISTS = 'UsernameExists',
+  USER_NOT_CONFIRMED = 'UserNotConfirmed',
+  USER_NOT_FOUND = 'UserNotFound',
+}
+
+export class UserManagementError extends Error {
```

Comment:
I seem to remember there being some limitations to extending `Error` -- did you confirm that `code` was being retained probably within a `catch` after a UserManagementError is thrown?

Reply:
yes

Reply:
Even after transpilation?

Reply:
I tested it earlier with the example controller and it worked as expected